### PR TITLE
Move some integration tests to the subtest framework, and run them in…

### PR DESCRIPTION
**Pull request by dlorenc**
… parallel.

### CODE REVIEW

**Comment by dlorenc** [comment]
@minikube-bot test this please

**Comment by dlorenc** [comment]
@r2d4 @aaron-prindle could you take a look?

File: test/integration/functional_test.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+	t.Run("DNS", testClusterDNS)
+	t.Run("EnvVars", testClusterEnv)
+	t.Run("Logs", testClusterLogs)
+	t.Run("SSH", testClusterSSH)
+	t.Run("Status", testClusterStatus)
+	t.Run("Addons", testAddons)
+	t.Run("Dashboard", testDashboard)
+	t.Run("ServicesList", testServicesList)
+}
```
Comment (r2d4) [inline test/integration/functional_test.go]: Do we need some teardown logic at the end?

Comment (dlorenc) [inline test/integration/functional_test.go]: We weren't deleting before. I kind of like it this way since it's easy to rerun them quickly.

Comment (r2d4) [inline test/integration/functional_test.go]: sgtm

**Comment by dlorenc** [comment]
I merged this but Github doesn't realize it for some reason. See 35acd21cb1d3a2281729eb37d6799971083e37e3

**Pull request closed without merging**